model_analyzer/config/generate/automatic_model_config_generator.py

```diff
@@ -24,12 +24,13 @@
 class AutomaticModelConfigGenerator(BaseModelConfigGenerator):
     """ Given a model, generates model configs in automatic search mode """
 
-    def __init__(self, config, model, client, variant_name_manager,
+    def __init__(self, config, gpus, model, client, variant_name_manager,
```

Comment:
I don't like to see our constructor arg list continue to grow. If it is a necessary change, then we may want to refactor. I know on c++ land we would create a struct for stuff like this. Maybe the common info goes into a ConfigOptions struct that contains most of this? Or we piggy-back off of the SearchConfig class I recently created?

My main question is, do we actually need the GPUs passed around everywhere?

Reply:
I still don't think that any of the generators will ever need GPUs. We can create the temporary triton server without GPUs.

Reply:
We can, but it doesn't result in the same config being created - at a minimum we'd need to hack in and change from CPU to GPU_ONLY. Not sure if there are any other side effects, this seems unnecessarily risky.